Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM python:3.9-slim-bullseye
+
+# Copy minimal set of SQLFluff install files
+# into their own folder for easier debugging.
+WORKDIR /app
+COPY src ./src
+COPY setup.py .
+COPY MANIFEST.in .
+COPY README.md .
+
+# Install SQLFluff in virtual environment to
+# avoid potential clashes with system dependencies.
+RUN python -m venv .venv \
+    && . .venv/bin/activate \
+    && pip install --upgrade pip setuptools wheel \
+    && pip install .
+
+# Create unpriveleged user.
+USER 5000
```

Comment:
What does this do? It doesn't appear to create a user if I understand correctly. Isn't it switching to a pre-existing user with UD 5000?

Reply:
So by default the docker containter runs as root which is convenient for development, however it's recommended to run as a non-root user for security (not really a concern for us but still good to follow best practices).

uid 5000 is in the range for normal users in debian, I just picked that number as that was what was used in the original issue so I assume that's a convention. You can see below that the container is successfully running as uid 5000:
![image](https://user-images.githubusercontent.com/80432516/142742426-36729b99-6fd2-463f-9b90-ff2f43e7c2f9.png)
Given the use case of this container there's no point creating a username etc and assigning it to uid 5000.

I'll change the the comment to say "Switch to non-root user" for clarity

Reply:
Cool. I didn't know this was possible. It's weird, but it's Linux, and if it works, that's good enough for me.